File: redsheet-portal/src/app/database-models/v-their-batna.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTheirBatna {
    constructor(vTheirBatna?: vTheirBatna )  {
        let properties = [ "projectRequirementId", "theirBatanValue", "theirBatnaId",];
        for (let property of properties)
            if (vTheirBatna && vTheirBatna[property])
                this[property] = vTheirBatna[property];
    }
 
	projectRequirementId : number =   0 ;
 
    @required()
    @maxLength(1000)
	theirBatanValue : string =   undefined;
 
    @range(1,2147483647)
	theirBatnaId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vTheirBatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTheirBatnas", Schema = "dbo")]
	[Serializable]
	public partial class vTheirBatna  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectRequirementId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TheirBatanValue { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int TheirBatnaId { get; set; } 

	public vTheirBatna()
	{
    }

    }
}

